imx cleanup

fboudra opened the pull request:
Clean up the imx makefile in order to make it more generic and reusable by other imx platforms.

jforissier commented on imx.mk:
```diff
@@ -136,7 +137,9 @@ $(FIRMWARE_PATH)/.unpacked: $(FIRMWARE_PATH)/$(FIRMWARE_BIN)
 	touch $(FIRMWARE_PATH)/.unpacked
-	
+	cp $(FIRMWARE_PATH)/$(FIRMWARE_VERSION)/firmware/ddr/synopsys/lpddr4_pmu_train_*.bin \
+		$(U-BOOT_PATH)
```
(Commit " imx.mk: copy the ddr firmwares and tf-a bl31 binray to U-Boot")
This is fragile. If `u-boot` is cleaned (removed and cloned again, or `git clean -f`), then `make u-boot` won't be able to build it again. I would prefer to have this in the u-boot build recipe.

fboudra replied:
ack, will do.

fboudra commented on imx.mk:
```diff
@@ -10,7 +10,9 @@ override COMPILE_S_KERNEL  := 64
-BR2_TARGET_GENERIC_GETTY_PORT = ttymxc0
+SERIAL_CONSOLE    ?= ttymxc0
+
+BR2_TARGET_GENERIC_GETTY_PORT = $(SERIAL_CONSOLE)
```
that's right, I was concerned about the buildroot naming scheme and coincidentally, longer to write variable. I'm fine to go with `BR2_TARGET_GENERIC_GETTY_PORT ?=` if it's preferred.

fboudra commented on imx.mk:
```diff
@@ -54,7 +54,7 @@ TF_A_EXPORTS = CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
 TF_A_FLAGS += BL32=$(ROOT)/optee_os/out/arm/core/tee-raw.bin
 
-tfa: optee-os u-boot
+tfa:
```
it looks like building tf-a without optee didn't trigger any error. I'll have another look.

fboudra replied:
nevermind, parallel build masked this issue. will fix.

fboudra commented on imx.mk:
```diff
@@ -56,6 +56,7 @@ TF_A_FLAGS += BL32=$(OPTEE_PATH)/out/arm/core/tee-raw.bin
 
 tfa:
 	$(TF_A_EXPORTS) $(MAKE) -C $(TF_A_PATH) $(TF_A_FLAGS) all fip
+	cp $(TF_A_PATH)/build/imx8mp/release/bl31.bin $(U-BOOT_PATH)
```
ack, will do.

jforissier commented on imx.mk:
```diff
@@ -133,7 +136,7 @@ $(FIRMWARE_PATH)/.unpacked: $(FIRMWARE_PATH)/$(FIRMWARE_BIN)
 	 chmod 711 $(FIRMWARE_BIN) && ./$(FIRMWARE_BIN) --auto-accept)
 	touch $(FIRMWARE_PATH)/.unpacked
-	
+
```
This could perhaps be moved to the "cosmetic" patch but no big deal

jforissier reviewed (commented):
- For commits:
  "imx.mk: allow to override OPTEE_OS_PLATFORM"
  "imx.mk: introduce TFA_PLATFORM"
  "imx.mk: allow the BR variables to be configured"
  "imx.mk: remove U-BOOT_BIN - unused"
  "imx.mk: remove FIRMWARE_BIN - duplicated"
  "imx.mk: reuse firmware version variable" (minor: could be "imx.mk: reuse FIRMWARE_VERSION")
  "imx.mk: group paths variables together"
  "imx.mk: remove tfa dependency on u-boot"
  "imx.mk: cosmetic: use tabs instead of spaces"
  "imx.mk: use paths variables as much as possible"
  "imx.mk: copy the ddr firmwares and tf-a bl31 binary to U-Boot"
  "imx.mk: add variables for defconfig and dt" (minor: "imx.mk: add variables for defconfig and DT")
  "imx: add variables to define imx-mkimage SoC"
  
  `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

- For "imx.mk: reorder all/clean targets": please see below

fboudra commented on imx.mk:
```diff
@@ -37,8 +37,8 @@ BOOT_IMG		?= $(ROOT)/out/boot.img
-all: tfa u-boot linux optee-os buildroot flash-image
-clean: tfa-clean buildroot-clean u-boot-clean optee-os-clean
+all: ddr-firmware optee-os tfa u-boot linux buildroot flash-image
+clean: ddr-firmware-clean optee-os-clean tfa-clean u-boot-clean buildroot-clean
```
done. Left all target with the same dependencies.

fboudra commented:
@jforissier thanks for the review. Updated the serie and added the reviewed-by tags except for the `imx.mk: reorder all/clean targets`. The patch has been updated though, based on your feedback.

jforissier commented on imx.mk:
```diff
@@ -37,8 +37,8 @@ BOOT_IMG		?= $(ROOT)/out/boot.img
 ################################################################################
-all: tfa u-boot linux optee-os buildroot flash-image
-clean: tfa-clean buildroot-clean u-boot-clean optee-os-clean
+all: optee-os tfa u-boot linux buildroot flash-image
```
The target order doesn't matter from a functional standpoint. If at least you were following some kind of convention (alphabetical order...) it could make sense, but as it is, not really :wink: 
I suggest to not touch this line.

jforissier commented on imx.mk:
```diff
@@ -37,8 +37,8 @@ BOOT_IMG		?= $(ROOT)/out/boot.img
-all: tfa u-boot linux optee-os buildroot flash-image
-clean: tfa-clean buildroot-clean u-boot-clean optee-os-clean
+all: optee-os tfa u-boot linux buildroot flash-image
+clean: ddr-firmware-clean optee-os-clean tfa-clean u-boot-clean buildroot-clean
```
That's OK, please keep this line only and rename the commit "imx.mk" add ddr-firmware-clean to clean target". With that it is `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`.

fboudra replied:
done

fboudra commented:
@jforissier gentle ping

jforissier commented:
@fboudra sorry I did not notice you pushed updates. I'm merging this now, many thanks!
